src/components/SelectionList/types.ts

```diff
@@ -0,0 +1,241 @@
+import type {ReactNode} from 'react';
+import type {GestureResponderEvent, InputModeOptions, StyleProp, TextStyle, ViewStyle} from 'react-native';
+import type {SubAvatar} from '@components/SubscriptAvatar';
+import type {Errors, PendingAction} from '@src/types/onyx/OnyxCommon';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+
+type CommonListItemProps<TItem> = {
+    /** Whether this item is focused (for arrow key controls) */
+    isFocused?: boolean;
+
+    /** Style to be applied to Text */
+    textStyles?: StyleProp<TextStyle>;
+
+    /** Style to be applied on the alternate text */
+    alternateTextStyles?: StyleProp<TextStyle>;
+
+    /** Whether this item is disabled */
+    isDisabled?: boolean;
+
+    /** Whether this item should show Tooltip */
+    showTooltip: boolean;
+
+    /** Whether to use the Checkbox (multiple selection) instead of the Checkmark (single selection) */
+    canSelectMultiple?: boolean;
+
+    /** Callback to fire when the item is pressed */
+    onSelectRow: (item: TItem) => void;
+
+    /** Callback to fire when an error is dismissed */
+    onDismissError?: (item: TItem) => void;
+};
+
+type User = {
+    /** Text to display */
+    text: string;
+
+    /** Alternate text to display */
+    alternateText?: string;
+
+    /** Key used internally by React */
+    keyForList: string;
+
+    /** Whether this option is selected */
+    isSelected?: boolean;
+
+    /** Whether this option is disabled for selection */
+    isDisabled?: boolean;
+
+    /** User accountID */
+    accountID?: number;
+
+    /** User login */
+    login?: string;
+
+    /** Element to show on the right side of the item */
+    rightElement: JSX.Element;
```

Comment:
Can it be ReactNode?
```suggestion
    rightElement: ReactNode
```

Reply:
Will be ReactElement